### PR TITLE
fix: prevent raise of exception on unparsable date fields

### PR DESCRIPTION
**Pull request by jsam**
This PR adds robustness to parsing of date-time objects in Project, DatasetTag, Dataset and DatasetFile.

### CODE REVIEW

File: tests/core/commands/test_serialization.py [hunk 1]
```diff
@@ -18,9 +18,18 @@
 """Serialization tests for renku models."""
 
 import datetime
-
 import yaml
 
+from renku.core.models.projects import Project
+
+
+def test_project_serialization(project_metadata):
```
[the remark this answers is not in the document]
Comment (jsam) [reply tests/core/commands/test_serialization.py]: You're right! Thank you. I will add it this existing parametric test `test_project_metadata_compatibility`

**Comment by rokroskar** [comment]
This is just going to break things later down the line I'm afraid. At some point some graph query is going to try to compare dates and it's going to fail in some strange way. With #767 and #809 we can verify on project creation that the metadata makes sense. What do you think?

**Comment by jsam** [comment]
Ofc, we should verify on project creation that the metadata makes sense and all of that makes total sense, but this PR is not about this, it more about not crashing the CLI while parsing unparsable datetime. Users can always change the metadata manually and this PR is about not screaming in case of failure, it doesn't change the behaviour, just adds robustness to the code.

Maybe I'm missing something. What would this break?

**Comment by rokroskar** [comment]
It's ignoring a bad value that will eventually end up in the KG. I'd rather fail early than in some unpredictable way much later in the processing/using this data.

**Comment by jsam** [comment]
@rokroskar Which is exactly the reason why we are adding the validation which will capture this. How is this related to writing defensive code and being robust?

**Comment by jsam** [comment]
Ok, as discussed let's put this on hold until shackl validation gets merged.

**Pull request closed without merging**